### DISCUSSION TITLE
【2021.9.15】community conference discussion

### DISCUSSION THREAD

**Discussion by fanux** [sealerio/sealer#586] [General]
# source code 
* cluster upgrade
* COPY command behavior
* add imageList feature to local build and cloud build
* Change  kubeadm API to V1beta3 when k8s version gt 1.23.0
* incremental updating
* Time sync checker, Docker checker
* image merge
* ulimit settings
* show more image details
* move some submodule to pkg dir
* unit test

# base image
* ARM base image
* GPU base image
* Containerd support & more kubernetes version & rootfs maker robot

# CNCF
* code of conduct
* website
* roadmap

# others
* lint markdown
* maintainer rules
* governance rules

**Comment by kakaZhou719** [comment]
lite build : support to load offline images .

**Reply by fanux** [reply]
@lllwan lite build not support cache docker images now?

**Reply by lllwan** [reply]
Lite build already includes the ability of image cache. Why do you need to add it in local build?

**Reply by kakaZhou719** [reply]
Sorry for the unclear description。i mean , could we support to import docker offline images and cache it directly?no need to run docker pull to fetch the desired images.

**Reply by lllwan** [reply]
fake docker pull request？

**Comment by kakaZhou719** [comment]
plugin : some application need to do post config after cluster running , while its hard for our plugin to support it.

for example: openebs cstor. it need to create `CStorPoolCluster` CRD  when this application in production ready, and some parameter was generated by the operator like `blockDeviceName`. 
  https://github.com/alibaba/sealer/blob/main/applications/openebs/cstor/cspc.yaml

**Reply by fanux** [reply]
Do you have some idea about how to get blockDeviceName?

**Reply by fanux** [reply]
I think this is openebs design weakness, not friendly to deploy...

**Comment by clcc2019** [comment]
How about apply dragonfly to sealer?

**Reply by clcc2019** [reply]
https://github.com/dragonflyoss/Dragonfly2

**Reply by fanux** [reply]
Build a dragonfly CloudImage?

**Reply by clcc2019** [reply]
> Build a dragonfly CloudImage?
dragonfly  feature
![image](https://user-images.githubusercontent.com/46099295/133358548-89497f4d-7245-447f-ba05-2641364973fe.png)

**Comment by cuisongliu** [comment]
rootfs robot need append subcommand to sealer ??

**Reply by fanux** [reply]
I think most user don't care about the robot, so needn't add it in sealer.   

But robot should support containerd, v1.20+, ARM...

**Comment by fanux** [comment]
1. What is sealer roadmap?
2. maintainer rules?
3. docs/tech and docs/community
4. unit test plan

**Comment by fanux** [comment]
![image](https://user-images.githubusercontent.com/8912557/133193471-832eeb8c-9d3a-4bc1-a936-5ad14647cfcf.png)

**Comment by myxxhui** [comment]
It would be perfect if Sealer could support base mirroring of GPU resources and sealer ClusterFile could designate some nodes as GPU nodes

**Comment by fanux** [comment]
COPY dir . dst is $rootfs/dir is better I think. = cp -r dir ./
COPY dirA ./dirB dst is $rootfs/dirB/dirA is better. = cp -r dirA ./dirB
COPY dirA /dirB dst is /dirB/dirA = cp -r dirA /dirB

**Comment by fanux** [comment]
* registry push

**Comment by fanux** [comment]
COPY ingress.tar /images

**Comment by fanux** [comment]
![image](https://user-images.githubusercontent.com/8912557/133879086-f13e3e37-65c3-43e2-977c-e8ebf8c8fb34.png)

**Comment by fanux** [comment]
ACK-D:
  1. 更好用的helm基础镜像 @kakaZhou719 
  2. 文档&官网建设  @bxy4543 
  3. Build模块重构，先写文档 @kakaZhou719

**Comment by fanux** [comment]
ssh client session close
